include/multipass/settings/basic_setting_spec.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2021-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef MULTIPASS_BASIC_SETTING_SPEC_H
+#define MULTIPASS_BASIC_SETTING_SPEC_H
+
+#include "setting_spec.h"
+
+namespace multipass
+{
+class BasicSettingSpec : public SettingSpec
+{
+public:
+    BasicSettingSpec(QString key, QString default_);
```

Comment:
This was sort of covered in the spec (sections UI->Set->(Domain|Resetting)), but here's a summary of how I see it:

> 1. Should we require a default?

A setting provides a way to configure some aspect of how Multipass behaves. But until the user chooses a behavior, Multipass must choose on its own. For that "default" behavior to be re-selectable, it must correspond to something that can be provided in `multipass set`. While this could be achieved with null, I think it's more informative to show an actual value on `multipass get`. For instance, it is better to see `multipass get local.driver` say `qemu` than `null`. Therefore...

> 2. Should settings be nullable by default?

... if this meant adding a null value to *every* setting's domain, in order to express the default (i.e. `get` really did say `null`), I think that would be artificial and inexpressive. In some cases, the apparent "absence of a value" has useful meaning (see below), so it makes sense for a corresponding value in the domain of the setting. But that is a case-by-case thing, and that value can't be assumed to be the setting's default. 

OTH, we could in principle use `null` only for resetting. That is, `null` would not correspond to an actual value and the domain of the setting would remain intact. Instead, it would act as a special signal for the `set` command, to designate whatever was the default for that setting. That could probably be achieved, but...

> 3. Does null equal "use default"?

... there are practical difficulties, as discussed in the spec: how to write that value? The empty string is already assigned to specific behavior (≠ reset) in some settings (e.g. to disable primary, hotkey). Other strings, like "null", are valid values, and quotes and escapes are problematic in the shell. BTW, we might as well use "reset" instead of "null" at that point. But I think we'd be better off using a flag to signal the special intent to `set` (e.g. `set --reset a.b.c`). I realize that would change the interpretation of the positional arguments, though. And we would need to replicate the same thing in any other settings UI (e.g. a special button in a GUI).

Anyway, the verdict in the doc was not to support resetting, so I kept what we had in this regard. I believe that would be a feature on its own, meriting a dedicated spec and PR.